fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java

```diff
@@ -75,9 +75,27 @@ public static IntLiteral dateDiff(LiteralExpr first, LiteralExpr second) throws
 
     @FEFunction(name = "date_add", argTypes = { "DATETIME", "INT" }, returnType = "DATETIME")
     public static DateLiteral dateAdd(LiteralExpr date, LiteralExpr day) throws AnalysisException {
-        Date d = new Date(getTime(date));
-        d = DateUtils.addDays(d, (int) day.getLongValue());
-        return new DateLiteral(DateFormatUtils.format(d, "yyyy-MM-dd HH:mm:ss"), Type.DATETIME);
+        DateLiteral dateLiteral = (DateLiteral) date;
+        DateTimeFormatterBuilder builder = new DateTimeFormatterBuilder();
+        if (dateLiteral.getType() == Type.DATE) {
+            builder.appendYear(4, 4).appendLiteral("-").
+                    appendMonthOfYear(2).appendLiteral("-").appendDayOfMonth(2);
+        } else {
+            builder.appendYear(4, 4).appendLiteral("-").
+                    appendMonthOfYear(2).appendLiteral("-")
+                    .appendDayOfMonth(2).appendLiteral(" ")
+                    .appendHourOfDay(2).appendLiteral(":")
+                    .appendMinuteOfHour(2).appendLiteral(":")
+                    .appendSecondOfMinute(2);
+        }
+        LocalDateTime dateTime = 
+            builder.toFormatter().parseLocalDateTime(dateLiteral.getStringValue()).plusDays((int) day.getLongValue());
+        if (dateLiteral.getType() == Type.DATE) {
+            return new DateLiteral(dateTime.getYear(), dateTime.getMonthOfYear(), dateTime.getDayOfMonth());
+        } else {
+            return new DateLiteral(dateTime.getYear(), dateTime.getMonthOfYear(), dateTime.getDayOfMonth(),
+                    dateTime.getHourOfDay(), dateTime.getMinuteOfHour(), dateTime.getSecondOfMinute());
```

Comment:
Why not move this logic to DateLiteral. And I think we can make formatter static which is thread-safe.